### PR TITLE
chore: allocated types, consts and utility functions to separate files

### LINKED ISSUE #277: Allocating a separate file for constants
## Description
The constants are kept in random files which could cause problems related to SSR.

## Solution
Allocate a separate file for constants

## Willing to solve the issue or working on it?
Yes

### PR DESCRIPTION
**Pull request by Re-Dei**
## Issue Fixed
#277 

## Changes:
Allocated the constants, types, and utility functions that were exported in the files below to separate files:
- app/api/leaderboard/functions.ts
- app/api/leaderboard/route.ts
- app/leaderboard/_components/Leaderboard.tsx
- app/leaderboard/_components/LeaderboardWrapper.tsx
- app/leaderboard/_components/Searchbar.tsx
- app/leaderboard/page.tsx
-  components/DateRangePicker.tsx
- components/contributors/LeaderboardCard.tsx
- components/contributors/TopContributor.tsx

## Problem solved:
Importing from these files could cause server-side rendering issues so I allocated them to separate files. Also, it has the added benefit of decreasing chunk size as importing from these files imported the react components with themselves.

## Testing:
Tested in both production and development in WSL2.

### CODE REVIEW

**Comment by Re-Dei** [comment]
@rithviknishad please review this PR.
Regarding GSOC, I know it's a bit early but I have a favor to ask. Could you check out my project proposal? Just a few minutes will be fine. I heard it was a common practice to get proposals checked by mentors a few weeks before application period begins. I am sorry if its not the case. If its ok by you, I will dm my google docs link to you in slack. Also in how many projects can each contributor apply? I didn't find any information related to it. I just know each contributor can apply to at most 3 organizations.

File: lib/const.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+export const TOP_CONTRIBUTOR_CATEGORIES = {
+  eod_update: "EOD Updates",
+  pr_opened: "Pull Requests Opened",
+  pr_merged: "Pull Requests Merged",
+  pr_reviewed: "Pull Requests Reviewed",
+  issue_opened: "Issues Opened",
+  comment_created: "Comments Created",
+};
```
Comment (rithviknishad) [inline lib/const.ts:27]: Add `as const` for all these, so that we can use it in a typesafe manner in other places

File: lib/types.ts [hunk 2]
```diff
@@ -126,3 +131,39 @@ export interface ReleasesResponse {
...
+  linkedin: string;
+  slack: string;
+};
+
+export type LeaderboardPageProps = {
+  searchParams: {
+    search?: string;
+    between?: string; // <start-date>...<end-date>
+    sortBy?: LeaderboardSortKey;
+    role?: ("core" | "intern" | "operations" | "contributor")[];
```
Comment (rithviknishad) [inline lib/types.ts:160]:
```suggestion
    role?: (keyof typeof FILTER_BY_ROLE_OPTIONS)[];
```

File: app/leaderboard/_components/Leaderboard.tsx [hunk 3]
```diff
@@ -3,12 +3,16 @@ import { TbZoomQuestion } from "react-icons/tb";
 import TopContributor from "../../../components/contributors/TopContributor";
 import { getWeekNumber, parseDateRangeSearchParam } from "@/lib/utils";
 // import { formatDate } from "@/components/DateRangePicker";
```
Comment (rithviknishad) [inline app/leaderboard/_components/Leaderboard.tsx:5]:
```suggestion
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Pull request merged**